system_files/desktop/shared/usr/libexec/bazzite-bluetooth-ds4-ds5-workaround

```diff
@@ -0,0 +1,185 @@
+#!/bin/bash
+
+# Script Name: bazzite-bluetooth-ds4-ds5-workaround
+# Author: https://github.com/thekk1
+# Version: 1.4
+#
+# Related issues:
+# https://github.com/ublue-os/bazzite/issues/1289
+# https://github.com/ValveSoftware/steam-for-linux/issues/8678
+#
+# Description:
+# This script monitors connected PS4 and PS5 gamepad devices and disconnects
+# the Bluetooth connection if specific button combinations are detected (Home and Triangle buttons).
+# The script handles background processes for each device and maintains a record of process IDs
+# to ensure proper cleanup of processes associated with devices that are no longer present.
+#
+# This script should be removed if there is a native solution under one of the mentioned issues.
+#
+# Usage:
+# ./bazzite-bluetooth-ds4-ds5-workaround [-d]
+# -d: Enable debug mode
+
+DEBUG=false
+MONITOR_FILE="/tmp/monitored_devices.txt"
+truncate -s 0 $MONITOR_FILE
+
+# Check for the debug flag
+while getopts "d" opt; do
+    case ${opt} in
+        d ) DEBUG=true ;;
+        \? ) echo "Usage: $0 [-d]" ;;
+    esac
+done
+
+# Function to find gamepad devices (PS4 and PS5)
+find_gamepad_devices() {
+    local devices=()
+    local base_path="/sys/class/input"
+
+    for entry in "$base_path"/event*; do
+        local device_path="$entry/device"
+        local uevent_path="$device_path/uevent"
+
+        if [[ -f "$uevent_path" ]]; then
+            local devicename
+            devicename=$(grep "NAME" "$uevent_path" | cut -d'=' -f2 | tr -d '"')
+            if [[ "$devicename" == "Wireless Controller" || "$devicename" == "DualSense Wireless Controller" ]]; then
+                devices+=("$(basename "$entry")")
+            fi
+        fi
+    done
+    echo "${devices[@]}"
+}
+
+# Function to get MAC address from uevent file
+get_mac_address_from_uevent() {
+    local device="$1"
+    local uniq_path="/sys/class/input/$device/device/uevent"
+
+    if [[ -f "$uniq_path" ]]; then
+        local mac_address
+        mac_address=$(grep "UNIQ" "$uniq_path" | cut -d'=' -f2 | tr -d '"')
+        echo "$mac_address"
+    else
+        echo ""
+    fi
+}
+
+# Function to monitor the gamepad and disconnect Bluetooth connection
+monitor_gamepad() {
+    local device_path="$1"
+    local bluetooth_mac_address="$2"
+    local process_name="monitor_gamepad_$bluetooth_mac_address"
+
+    $DEBUG && echo "Starting monitoring on device: $device_path"
+
+    # Check if process is already running
+    if pgrep -f "$process_name" > /dev/null; then
+        $DEBUG && echo "Process for $bluetooth_mac_address is already running."
+        return
+    fi
+
+    # Define the command to disconnect all controllers
+    disconnect_cmd="for controller in \$(bluetoothctl list | awk '/Controller/ {print \$2}'); do bluetoothctl select \$controller; bluetoothctl disconnect $bluetooth_mac_address; done"
```

Comment:
this will not work, i fell into the same trap myself when i tried to automate operations through bluetoothctl using a script.
`bluetoothctl select` will select a controller then `exit`, this causes `bluetoothctl disconnect` to again potentially target the wrong/first controller

the only surefire way i managed to make it work in a script was to pipe the commands into `bluetoothctl` or feed them to it in another way like i showed in my comment :)

Using this will work (and i tested+verified it)
```bash
disconnect_cmd="for controller in \$(bluetoothctl list | awk '/Controller/ {print \$2}'); do echo -e \"select \$controller\ndisconnect $bluetooth_mac_address\n\" | bluetoothctl; done"
```

everything else works though and its sweet! Thanks a bunch!

Reply:
All-right, thanks for your commitment. I made the purposed change. LGTM :smiley: